File: Microservices/IoT.House.Automation.Microservices.Arduino/IoT.House.Automation.Microservices.Arduino.Application/Interfaces/IEventBus.cs
using System;
using System.Collections.Generic;
using System.Text;
using IoT.House.Automation.Microservices.Arduino.Application.MessageBroker.Events;

namespace IoT.House.Automation.Microservices.Arduino.Application.Interfaces
{
    public interface IEventBus
    {
        void Publish(Event @event);

        void Subscribe<TEvent, TEventHandler>()
            where TEvent : Event
            where TEventHandler : IEventHandler<TEvent>;

        void Unsubscribe<TEvent, TEventHandler>()
            where TEvent : Event
            where TEventHandler : IEventHandler<TEvent>;
    }
}

File: Microservices/IoT.House.Automation.Microservices.Notification/src/domain/interfaces/IEventBus.ts
import { Event } from "../entities/events/event";
import { IEventHandler } from "./IEventHandler";

export interface IEventBus {
    publish(event : Event) : void;
    subscribe<TEventHandler extends IEventHandler<Event>>(event: string, eventHandler: TEventHandler) : void;
    unsubscribe(event: string) : void;
}